src/eduid/webapp/personal_data/views.py

```diff
@@ -82,6 +86,65 @@ def update_personal_data(
     personal_data = personal_data_user.to_dict()
     return success_response(payload=personal_data, message=PDataMsg.save_success)
 
+@pd_views.route("/user/name", methods=["POST"])
+@UnmarshalWith(UserNameRequestSchema)
+@MarshalWith(UserNameResponseSchema)
+@require_user
+def update_user_name(
+    user: User, given_name: str, surname: str, chosen_given_name: str | None = None
+) -> FluxData:
+    personal_data_user = PersonalDataUser.from_user(user, current_app.private_userdb)
+    current_app.logger.debug(f"Trying to save user {user}")
+
+    # disallow change of first name, surname if the user is verified
+    if not user.identities.is_verified:
+        personal_data_user.given_name = given_name
+        personal_data_user.surname = surname
+
+    # set chosen given name to either given name or a subset of given name if supplied
+    # also allow to set chosen given name to None
+    if (
+        chosen_given_name is not None
+        and is_valid_chosen_given_name(personal_data_user.given_name, chosen_given_name) is False
+    ):
+        return error_response(message=PDataMsg.chosen_given_name_invalid)
+
+    # mypy borked?
+    # error: Incompatible types in assignment (expression has type "str | None", variable has type "str")
+    personal_data_user.chosen_given_name = chosen_given_name
+
+    try:
+        save_and_sync_user(personal_data_user)
+    except UserOutOfSync:
+        return error_response(message=CommonMsg.out_of_sync)
+    current_app.stats.count(name="personal_data_saved", value=1)
```

Comment:
I think we should add more specific statistics now,
`current_app.stats.count(name="user_name_saved", value=1)`